Use system compiler builtins on BSD; retain upstream linkage elsewhere.

Index: codex-rs/modcargo-crates/v8-150.4.0/build/config/clang/BUILD.gn
--- codex-rs/modcargo-crates/v8-150.4.0/build/config/clang/BUILD.gn.orig
+++ codex-rs/modcargo-crates/v8-150.4.0/build/config/clang/BUILD.gn
@@ -243,7 +243,12 @@
 
 # Adds a dependency on the Clang runtime library clang_rt.builtins.
 clang_lib("compiler_builtins") {
-  if (is_mac) {
+  if (is_bsd) {
+    # Since there's no Rust in the toolchain, there's no concern that we'll use
+    # the Rust stdlib's intrinsics here.
+    #
+    # Don't define libname which makes this target do nothing.
+  } else if (is_mac) {
     libname = "osx"
   } else if (is_ios) {
     if (target_platform == "iphoneos") {
